entity-framework/core/what-is-new/ef-core-3.0/index.md

```diff
@@ -1,14 +1,199 @@
+
+Similarly to the interception features that existed in EF 6, interceptors allow you to intercept operations before or after they happen. When you intercept them before they happen, you are allowed to by-pass execution and supply alternate results from the interception logic. 
+
+For example, to manipulate command text, you can create an `IDbCommandInterceptor`: 
+
+``` csharp 
+public class HintCommandInterceptor : DbCommandInterceptor
+{
+  public override InterceptionResult ReaderExecuting(
+    DbCommand command, 
+    CommandEventData eventData, 
+    InterceptionResult result)
+  {
+    // Manipulate the command text, etc. here...
+    command.CommandText += " OPTION (OPTIMIZE FOR UNKNOWN)";
+    return result;
+  }
+}
+``` 
+
+And register it with your `DbContext`: 
+
+``` csharp
+services.AddDbContext(b => b
+  .UseSqlServer(connectionString)
+  .AddInterceptors(new HintCommandInterceptor()));
+```
+
+## Reverse engineering of database views
+
+Entity types without keys (previously known as [query types](xref:core/modeling/keyless-entity-types)) represent data that can be read from the database, but cannot be updated.
+This characteristic makes them an excellent fit for mapping database views in most scenarios, so we automated the creation of entity types without keys when reverse engineering database views.
+
+For example, using the [dotnet ef command-line tool](xref:core/miscellaneous/cli/dotnet) you can type:
+
+``` console
+dotnet ef dbcontext scaffold "Server=(localdb)\mssqllocaldb;Database=Blogging;Trusted_Connection=True;" Microsoft.EntityFrameworkCore.SqlServer
+```
+
+And the tool will now automatically scaffold types for views and tables without keys:
+
+``` csharp
+protected override void OnModelCreating(ModelBuilder modelBuilder)
+{
+  modelBuilder.Entity<Names>(entity =>
+  {
+    entity.HasNoKey();
+    entity.ToView("Names");
+  });
+
+  modelBuilder.Entity<Things>(entity =>
+  {
+    entity.HasNoKey();
+  });
+}
+```
+
+## Dependent entities sharing the table with the principal are now optional
+
+Starting with EF Core 3.0, if `OrderDetails` is owned by `Order` or explicitly mapped to the same table, it will be possible to add an `Order` without an `OrderDetails` and all of the `OrderDetails` properties, except the primary key will be mapped to nullable columns.
+
+When querying, EF Core will set `OrderDetails` to `null` if any of its required properties doesn't have a value, or if it has no required properties besides the primary key and all properties are `null`.
+
+``` csharp
+public class Order
+{
+    public int Id { get; set; }
+    public int CustomerId { get; set; }
+    public OrderDetails Details { get; set; }
+}
+
+[Owned]
+public class OrderDetails
+{
+    public int Id { get; set; }
+    public string ShippingAddress { get; set; }
+}
+```
+
+## EF 6.3 on .NET Core
```

Comment:
Should this be here in what is new in EFCore 3.0 page?

Reply:
I think it complements the information about EF Core 3.0 features for the reader. We can add something to the paragraph to convey that, "yeah, we know this is not really an EF Core 3.0 feature"

Reply:
Made it smaller, with just a link to the EF 6.3 what's new page.